vector/src/main/scala/geotrellis/vector/reproject/package.scala

```diff
@@ -44,11 +44,11 @@ package object reproject {
       )
     }
 
-    def apply(extent: Extent, src: CRS, dest: CRS): Extent = {
+    def apply(extent: Extent, src: CRS, dest: CRS): Polygon = {
       val f = Transform(src, dest)
       val sw = f(extent.xmin, extent.ymin)
       val ne = f(extent.xmax, extent.ymax)
-      Extent(sw._1,sw._2,ne._1,ne._2)
+      Extent(sw._1,sw._2,ne._1,ne._2).toPolygon
```

Comment:
This needs to not only be a Polygon, but needs to actually reproject each point of the polygon.

```
def apply(extent: Extent, src: CRS, dest: CRS): Polygon = 
  apply(extent.toPolygon, src, dest)
```

would work.